src/utils/wallet/providers/keystone/index.ts

```diff
@@ -0,0 +1,272 @@
+    address: string | undefined;
+    publicKeyHex: string | undefined;
+    mfp: string | undefined;
+};
+
+export class KeystoneWallet extends WalletProvider {
+    private keystoneWaleltInfo: KeystoneWalletInfo | undefined;
+    private viewSdk: typeof sdk;
+    private dataSdk: KeystoneSDK;
+
+    constructor() {
+        super();
+        sdk.bootstrap();
+        this.viewSdk = sdk;
+        this.dataSdk = new KeystoneSDK({
+            origin: "babylon staking app"
+        });;
+    }
+
+    async connectWallet(): Promise<this> {
+        const keystoneContainer = this.viewSdk.getSdk();
+        const decodedResult = await keystoneContainer.read(
+            [SupportedResult.UR_CRYPTO_ACCOUNT],
+            {
+                title: "Sync Keystone with Babylon Staking App",
+                description: "Please scan the QR code displayed on your Keystone, Currently Only the first Taproot Address will be used",
+                renderInitial: {
+                    walletMode: "btc",
+                    link: "",
+                    description: [
+                        '1. Turn on your Keystone 3 with BTC only firmware.',
+                        '2. Click connect software wallet and use "Sparrow" for connection.',
+                        '3. Press the "Scan Keystone" button and scan the QR Code displayed on your Keystone hardware wallet',
+                        '4. The first Taproot address will be used for staking.'
+                    ],
+
+                },
+                URTypeErrorMessage:
+                    "The scanned QR code is not the sync code from the Keystone hardware wallet. Please verify the code and try again.",
+            }
+        );
+
+        if (decodedResult.status === ReadStatus.success) {
+            const accountData = this.dataSdk.parseAccount(decodedResult.result);
+            let xpub = accountData.keys[3].extendedPublicKey;
+            this.keystoneWaleltInfo = {
+                mfp: accountData.masterFingerprint,
+                extendedPublicKey: xpub,
+                path: accountData.keys[3].path,
+                address: undefined,
+                publicKeyHex: undefined,
+            }
+
+            if (this.keystoneWaleltInfo.extendedPublicKey) {
+                const { address, pubkeyHex } = generateBitcoinAddressFromXpub(this.keystoneWaleltInfo.extendedPublicKey, "M/0/0", bitcoin.networks.testnet);
+                this.keystoneWaleltInfo.address = address;
+                this.keystoneWaleltInfo.publicKeyHex = pubkeyHex;
+            }
+        } else {
+            throw new Error("Error reading QR code, Please try again.")
+        }
+
+        return (this)
+
+    }
+    async getWalletProviderName(): Promise<string> {
+        return "Keystone";
+    }
+
+    async getAddress(): Promise<string> {
+        if (this.keystoneWaleltInfo?.address) {
+            return this.keystoneWaleltInfo?.address;
+        }
+        throw new Error("Could not retrieve the address");
```

Comment:
```suggestion
        if (!this.keystoneWaleltInfo?.address) {
            throw new Error("Get Address error.");
        }
        return this.keystoneWaleltInfo?.address;
```

nit: think it's cleaner this way, up to you